src/com/amazon/ion/benchmark/schema/ReparsedType.java

```diff
@@ -0,0 +1,111 @@
+package com.amazon.ion.benchmark.schema;
+
+import com.amazon.ion.IonStruct;
+import com.amazon.ion.IonType;
+import com.amazon.ion.IonValue;
+import com.amazon.ion.benchmark.schema.constraints.*;
+import com.amazon.ionschema.Type;
+
+import java.util.HashMap;
+import java.util.Map;
+
+// Parsing the type definition in ISL file into ReparsedType format which allows getting constraints information directly.
+public class ReparsedType {
+    public final Type type;
+    private static final String KEYWORD_TIMESTAMP_PRECISION = "timestamp_precision";
+    private static final String KEYWORD_TYPE = "type";
+    private static final String KEYWORD_CODE_POINT_LENGTH = "codepoint_length";
+    private static final String KEYWORD_REGEX = "regex";
+    private static final String KEYWORD_CONTAINER_LENGTH = "container_length";
+    private static final String KEYWORD_BYTE_LENGTH = "byte_length";
+    private static final String KEYWORD_SCALE = "scale";
+    private static final String KEYWORD_PRECISION = "precision";
+    private static final String KEYWORD_VALID_VALUES = "valid_values";
+    private static final String KEYWORD_NAME = "name";
+    // Using map to avoid processing the multiple repeat constraints situation.
+    private final Map<String, ReparsedConstraint> constraintMap;
+
+    /**
+     * Initializing the newly created ReparsedType object.
+     * @param type represents type definition of ISL file.
+     */
+    public ReparsedType(Type type) {
+        this.type = type;
+        constraintMap = new HashMap<>();
+        getIsl().forEach(this::handleField);
+    }
+
+    /**
+     * Get the name of type definition.
+     * @return the name of type definition.
+     */
+    public String getName() {
+        return type.getName();
+    }
+
+    /**
+     * Handling the fields which are not used for specifying generated data.
+     * @param field represents the field contained by the type definition.
+     */
+    private void handleField(IonValue field) {
+        switch (field.getFieldName()) {
+            case KEYWORD_NAME:
+            case KEYWORD_TYPE:
+                return;
+            default:
```

Comment:
> This could also include fields that are open content and which would not be part of ISL constraints. Adding a comment `toConstraint` method for this would work. [EDIT]: As per offline discussion, there is still a recursive type definition change to be addressed along with this one before merging this.

The comment will be added in the next commit. For the recursive type definition change, there might be more potential updates along with this change. I will throw an issue for this comment and contain this change in the future PR.